api/app.js

```diff
@@ -9,14 +9,10 @@ module.exports = function(configFilePath) {
   var bodyParser = require('body-parser');
   var logger = require('./lib/log/logger');
   var HttpStatus = require('http-status-codes');
-  var oauth = require('./lib/oauth/oauth')(configFilePath);
-
-  if (!configFilePath || !fs.existsSync(configFilePath)) {
-      logger.error("Invalid configuration file path: " + configFilePath);
-      throw new Error('configuration file does not exist:' + configFilePath);
-  }
-  var settings = require(path.join(__dirname, './lib/config/setting.js'))((JSON.parse(
-        fs.readFileSync(configFilePath, 'utf8'))));
+  
+  var oauth = require('./lib/oauth/oauth')(settings);
+  var models = require('./lib/models')(settings.db, callback);
```

Comment:
It is recommend to do setting.validate first, then using the setting to initialize the models